File: backend/models/character.go
package models

import (
	"encoding/json"
	"time"
)

type Character struct {
	ID                string          `json:"id"`
	UserID            string          `json:"user_id"`
	Name              string          `json:"name"`
	Lineage           string          `json:"lineage"`
	Heritage          string          `json:"heritage"`
	Class             string          `json:"class"`
	Level             int             `json:"level"`
	Background        string          `json:"background"`
	Alignment         string          `json:"alignment"`
	XP                int             `json:"xp"`
	LuckPoints        int             `json:"luck_points"`
	STR               int             `json:"str_score"`
	DEX               int             `json:"dex_score"`
	CON               int             `json:"con_score"`
	INT               int             `json:"int_score"`
	WIS               int             `json:"wis_score"`
	CHA               int             `json:"cha_score"`
	HPMax             int             `json:"hp_max"`
	HPCurrent         int             `json:"hp_current"`
	HPTemp            int             `json:"hp_temp"`
	SavingThrowProfs  []string        `json:"saving_throw_profs"`
	SkillProfs        []string        `json:"skill_profs"`
	SkillExpertise    []string        `json:"skill_expertise"`
	PersonalityTraits string          `json:"personality_traits"`
	Ideals            string          `json:"ideals"`
	Bonds             string          `json:"bonds"`
	Flaws             string          `json:"flaws"`
	Backstory         string          `json:"backstory"`
	Equipment         json.RawMessage `json:"equipment"`
	Attacks           json.RawMessage `json:"attacks"`
	Spells            json.RawMessage `json:"spells"`
	Talents           json.RawMessage `json:"talents"`
	Features          json.RawMessage `json:"features"`
	CP                int             `json:"cp"`
	SP                int             `json:"sp"`
	EP                int             `json:"ep"`
	GP                int             `json:"gp"`
	PP                int             `json:"pp"`
	CreatedAt         time.Time       `json:"created_at"`
	UpdatedAt         time.Time       `json:"updated_at"`
}

File: frontend/src/types/character.ts
export interface Attack {
  name: string
  attackBonus: number
  damageDice: string
  damageType: string
  notes?: string
}

export interface SpellSlots {
  [level: string]: { max: number; used: number }
}

export interface SpellData {
  castingAbility?: string
  saveDC?: number
  attackBonus?: number
  slots: SpellSlots
  known: string[]
  prepared: string[]
}

export interface Character {
  id: string
  user_id: string
  name: string
  lineage: string
  heritage: string
  class: string
  level: number
  background: string
  alignment: string
  xp: number
  luck_points: number
  str_score: number
  dex_score: number
  con_score: number
  int_score: number
  wis_score: number
  cha_score: number
  hp_max: number
  hp_current: number
  hp_temp: number
  saving_throw_profs: string[]
  skill_profs: string[]
  skill_expertise: string[]
  personality_traits: string
  ideals: string
  bonds: string
  flaws: string
  backstory: string
  equipment: unknown[]
  attacks: Attack[]
  spells: SpellData
  talents: unknown[]
  features: string[]
  cp: number
  sp: number
  ep: number
  gp: number
  pp: number
  created_at: string
  updated_at: string
}

export interface RulesClass {
  id: string
  name: string
  hitDie: number
  saves: string[]
  spellcasting: boolean
  spellcastingAbility?: string
  description: string
  level1Features: string[]
}

export interface RulesLineage {
  id: string
  name: string
  size: string
  speed: number
  darkvision: boolean
  darkvisionRange?: number
  description: string
  traits: string[]
}

export interface RulesHeritage {
  id: string
  name: string
  description: string
  features: string[]
}

export interface RulesBackground {
  id: string
  name: string
  description: string
  skillChoices: string[]
  numSkillChoices: number
  talents: string[]
  equipment: string
}

export interface WizardData {
  name: string
  lineage: string
  heritage: string
  class: string
  background: string
  alignment: string
  skills: string[]
  talent: string
  scores: Record<string, number>
  personalityTraits: string
  ideals: string
  bonds: string
  flaws: string
  backstory: string
}
